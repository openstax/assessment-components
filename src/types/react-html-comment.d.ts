declare module 'react-html-comment';
