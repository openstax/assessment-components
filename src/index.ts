export * from './components/Answer';
export * from './components/AnswersTable';
export * from './components/CompletionStatus';
export * from './components/Question';
export * from './components/FreeResponseInput';
export * from './components/Card';
export * from './components/Exercise';
export * from './components/ProgressBar';
export * from './components/Loader';
export * from './components/ExerciseToolbar';
