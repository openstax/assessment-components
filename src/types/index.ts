export interface ExercisePersonData {
  user_id: ID;
  name: string;
}

export interface ExerciseAnswerData {
  id: ID;
  content_html: string;
  correctness?: string;
  feedback_html?: string;
  isCorrect?: boolean;
}

export interface ExerciseTagData {
  id: ID;
  type: string;
  is_visible: boolean;
  data: string;
}

export interface ExerciseQuestionData {
  id: ID;
  is_answer_order_important: boolean;
  stimulus_html: string;
  stem_html: string;
  answers: ExerciseAnswerData[];
  formats: ExerciseFormat[];
  collaborator_solutions?: CollaboratorSolution[];
}

export interface ExerciseQueryData {
  title: string;
  exercises: ExerciseData[];
}

export interface ExerciseData {
  tags: string[];
  uuid: string;
  group_uuid: string;
  number: ID;
  version: number;
  uid: ExerciseUid;
  published_at: string;
  authors: ExercisePersonData[];
  copyright_holders: ExercisePersonData[];
  derived_from: any[]; // eslint-disable-line @typescript-eslint/no-explicit-any
  is_vocab: boolean;
  context?: string;
  stimulus_html: string;
  questions: ExerciseQuestionData[];
  versions: number[];
  solutions_are_public: boolean;
}

export type Answer = {
  id: ID;
  correctness?: string | null | undefined;
  isCorrect?: boolean;
  content_html: string;
  selected_count?: number;
  feedback_html?: string;
};

export type Step = {
  type: 'exercise';
  task: Task;
  uid: string;
  id: number;
  available_points: AvailablePoints;
  preview: string;
  is_completed: boolean;
  answer_id?: ID;
  answer_id_order: ID[];
  free_response: '';
  feedback_html: '';
  correct_answer_id: ID;
  correct_answer_feedback_html: string;
  last_completed_at: Date;
  response_validation?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  external_url: '';
  formats?: ExerciseFormat[];
  can_be_updated: boolean;
  is_feedback_available: boolean;
  exercise_id: ID
  attempts_remaining: number;
  attempt_number: number;
  solution?: Solution;
  incorrectAnswerId: ID;
}

interface Solution {
  content_html: string;
  solution_type: string;
}

export type Task = {
  is_deleted: boolean;
  type?: 'homework';
};

type CollaboratorSolution = {
  content_html: string;
};

export type AnswerDisplayType = 'teacher-review' | 'teacher-preview' | 'student' | 'student-mpp';
export type ChosenAnswer = (ID | undefined)[];

export type ID = string | number;

type ExerciseUid = `${number}@${number}`;
type ExerciseFormat = 'multiple-choice' | 'free-response' | 'true-false';

export type AvailablePoints = `${number}.${number}`;
