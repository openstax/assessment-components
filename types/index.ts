export type answerType = {
  id: idType;
  correctness: string | null;
  isCorrect: boolean;
  content_html: string;
  selected_count?: number;
};

export type chosenAnswerType = (idType | undefined)[];

export type idType = string | number;
