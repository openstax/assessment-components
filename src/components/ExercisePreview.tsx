import React from "react";
import { ExerciseData, ExerciseQuestionData, StepBase } from "src/types";
import { Exercise } from "./Exercise";
import styled from "styled-components";


const StyledExercise = styled(Exercise)<{ showAllFeedback?: boolean; showCorrectAnswer?: boolean }>`
    ${({ showAllFeedback }) => !showAllFeedback &&
    `.step-card-footer {
        display: none;
      }`
    }
    ${({ showCorrectAnswer }) => !showCorrectAnswer &&
    `.answer-answer {
        font-weight: normal;
        & > div {
          display: none;
        }
      }`
    }
  `;
export interface ExercisePreviewProps {
  exercise: ExerciseData;
  selected?: boolean;
  showAllFeedback?: boolean;
  showCorrectAnswer?: boolean;
  overlayChildren?: React.ReactNode;
}

export const ExercisePreview = (
  {
    exercise,
    selected,
    showAllFeedback = false,
    showCorrectAnswer = false,
    overlayChildren,
  }: ExercisePreviewProps) => {

  const hideAnswerFeedback = (exercise: ExerciseData) => {
    exercise.questions.map(question =>
      question.answers.map(a => {
        a.feedback_html = '';
        a.correctness = showCorrectAnswer ? a.correctness : undefined;
      }));
    return exercise;
  };

  const exercisePreviewProps = (exercise: ExerciseData) => {
    const formatAnswerData = (questions: ExerciseQuestionData[]) => questions.map((q) => (
      {
        id: q.id,
        correct_answer_id: (q.answers.find((a) => a.correctness === '1.0')?.id || ''),
        content_html:
          showAllFeedback &&
          q.collaborator_solutions?.find(solution => solution.solution_type === 'detailed')?.content_html,
      }
    ));

    const questionStateFields = formatAnswerData(exercise.questions).reduce((acc, answer) => {
      const { id, correct_answer_id, content_html } = answer;
      return {
        ...acc,
        [id]: {
          answer_id: '',
          correct_answer_id,
          is_completed: showCorrectAnswer,
          solution: {
            content_html,
          }
        }
      };
    }, {});

    const step: StepBase = {
      id: 1,
      uid: exercise.uid,
      available_points: '1.0',
    };

    return {
      canAnswer: true,
      needsSaved: true,
      hasMultipleAttempts: false,
      onAnswerChange: () => undefined,
      onAnswerSave: () => undefined,
      onNextStep: () => undefined,
      apiIsPending: false,
      canUpdateCurrentStep: false,
      step: step,
      questionNumber: exercise.number as number,
      numberOfQuestions: exercise.questions.length,
      questionStates: questionStateFields,
      show_all_feedback: showAllFeedback,
    };
  };

  return (
    <StyledExercise
      exercise={showAllFeedback ? exercise : hideAnswerFeedback(exercise)}
      className={`preview-card ${selected ? 'is-selected' : ''}`}
      previewMode
      overlayChildren={overlayChildren}
      {...exercisePreviewProps(exercise)}
      showAllFeedback={showAllFeedback}
      showCorrectAnswer={showCorrectAnswer}
    />
  );
};
