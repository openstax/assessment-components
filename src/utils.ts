import { Answer, ID } from '../src/types';

export const ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';

const MAX_CORRECTNESS = '1.0';

export const isAnswerCorrect = function(answer: Answer, correctAnswerId?: ID | null) {
  // if answer does not have an id, check the isCorrect property.
  if (!(answer.id || correctAnswerId)) {
    return answer.isCorrect;
  }
  let isCorrect = answer.id === correctAnswerId;
  if (answer.correctness != null) { isCorrect = (answer.correctness === MAX_CORRECTNESS); }

  return isCorrect;
};

export const isAnswerIncorrect = (answer: Answer, incorrectAnswerId?: ID) =>
  // Allow multiple attempts to show incorrectness without the correct_answer_id
  answer.id === incorrectAnswerId;

export const isAnswerChecked = (answer: Answer, answerId?: ID) =>
   answer.id == answerId;

export function countWords(text: string) {
  const trimmedText = text.trim();
  //https://css-tricks.com/build-word-counter-app/
  const words = trimmedText.match(/\b[-?(\w+)?]+\b/gi);
  if(!words) return 0;
  return words.length;
}

export const numberfyId = (id: ID) => typeof id === 'string' ? parseInt(id, 10) : id;
