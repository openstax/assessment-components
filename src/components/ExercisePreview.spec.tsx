import { ExercisePreview } from './ExercisePreview';
import renderer from 'react-test-renderer';
import { ExerciseData, ID, QuestionState } from 'src/types';

describe('ExercisePreview', () => {
  describe('using step data', () => {

    let exercise: ExerciseData;
    let questionStates: { [key: ID]: QuestionState };

    beforeEach(() => {
      exercise = {
        uid: '1@1',
        uuid: 'e4e27897-4abc-40d3-8565-5def31795edc',
        group_uuid: '20e82bf6-232e-40c8-ba68-2d22c6498f69',
        number: 1,
        version: 1,
        published_at: '2022-09-06T20:32:21.981Z',
        context: 'Context',
        stimulus_html: '<b>Stimulus HTML</b>',
        tags: [],
        authors: [{ user_id: 1, name: 'OpenStax' }],
        copyright_holders: [{ user_id: 1, name: 'OpenStax' }],
        derived_from: [],
        is_vocab: false,
        solutions_are_public: false,
        versions: [1],
        questions: [{
          id: '1234@5',
          collaborator_solutions: [],
          formats: ['true-false'],
          stimulus_html: '',
          stem_html: '',
          is_answer_order_important: false,
          answers: [{
            id: '1',
            correctness: '1.0',
            content_html: 'True',
          }, {
            id: '2',
            correctness: undefined,
            content_html: 'False',
          }],
        }],
      };
      questionStates = {
        '1234@5': {
          available_points: '1.0',
          is_completed: true,
          answer_id_order: ['1', '2'],
          answer_id: undefined,
          free_response: 'Free response',
          feedback_html: 'Feedback',
          correct_answer_id: '1',
          correct_answer_feedback_html: 'Feedback for the correct answer',
          attempts_remaining: 0,
          attempt_number: 1,
          incorrectAnswerId: 0,
          canAnswer: false,
          needsSaved: false,
          apiIsPending: false,
        },
      };
    });

    it.each`
        selected | description                          | overlay                  | showFeedback
        ${true}  | ${'with overlay and selected true'}  | ${<button>Over</button>} | ${undefined}
        ${true}  | ${'with overlay and selected false'} | ${<button>Over</button>} | ${false}
        ${false} | ${'without overlay'}                 | ${undefined}             | ${true}
    `('matches snapshot %description', (
      { selected, overlay, showFeedback }: { selected: boolean, overlay: JSX.Element, showFeedback: boolean }) => {
      const tree = renderer.create(
        <ExercisePreview
          exercise={exercise}
          selected={selected}
          overlayChildren={overlay}
          showAllFeedback={showFeedback}
          questionStates={selected ? questionStates : undefined}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
