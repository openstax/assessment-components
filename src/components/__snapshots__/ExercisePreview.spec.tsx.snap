// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExercisePreview using step data matches snapshot with overlay and correctness 1`] = `
<div
  className="sc-ksZaOG gLzA-Dq"
>
  <div
    className="sc-gsnTZi dxdISE"
    data-task-step-id={1}
  >
    <div
      className="sc-bczRLJ eAXREJ exercise-step sc-breuTD dSaRVj sc-hAZoDl sc-fEOsli fZGyZE iCBiYQ preview-card is-selected"
    >
      <div
        onBlur={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        tabIndex={0}
      >
        <div
          id="step-card"
        >
          <div
            className="sc-dkzDqf eYRLRg step-card-header"
          >
            <div>
              <h2
                className="question-info"
              >
                <span>
                  Question 1
                </span>
                <span
                  className="separator"
                >
                  |
                </span>
                <span
                  className="question-id"
                >
                  ID: 
                  1@1
                </span>
              </h2>
            </div>
          </div>
          <div
            className="sc-hKMtZM bwRTaS"
          >
            <div>
              <div
                className="step-card-body exercise-context"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Context",
                  }
                }
              />
              <div
                className="step-card-body exercise-stimulus"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<b>Stimulus HTML</b>",
                  }
                }
              />
              <div
                data-test-id="student-exercise-question"
              >
                <div
                  className="sc-iBkjds bohpoA openstax-question step-card-body has-correct-answer"
                  data-question-number={1}
                  data-test-id="question"
                >
                  <div
                    className="answers-table"
                  >
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled answer-selected answer-correct"
                      >
                        <input
                          checked={true}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-0"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-0"
                        >
                          <span
                            aria-label="Selected Choice A:"
                            className="answer-letter-wrapper"
                            data-answer-choice="A"
                            data-test-id="answer-choice-A"
                          />
                          <div
                            className="answer-answer"
                          >
                            <div
                              className="sc-gKXOVf QPaPs"
                            >
                              Correct
                               Answer
                            </div>
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "True",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled"
                      >
                        <input
                          checked={false}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-1"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-1"
                        >
                          <span
                            aria-label="Choice B:"
                            className="answer-letter-wrapper"
                            data-answer-choice="B"
                            data-test-id="answer-choice-B"
                          />
                          <div
                            className="answer-answer"
                          >
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "False",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                  </div>
                  <div
                    className="detailed-solution"
                  >
                    <div
                      className="header"
                    >
                      Detailed solution:
                    </div>
                    <div
                      className="solution"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<span data-math='e^{\\\\text{Blue}}'></span><span data-math='e^{\\\\text{Blue}}'></span>",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="sc-ftvSup fUvOut step-card-footer"
                >
                  <div
                    className="step-card-footer-inner"
                  >
                    <div
                      className="points"
                      role="status"
                    >
                      <span
                        className="attempts-left"
                      />
                    </div>
                    <div
                      className="controls"
                    >
                      <button
                        className="sc-jSMfEi hsxEPT"
                        data-test-id="submit-answer-btn"
                        disabled={true}
                        onClick={[Function]}
                      >
                        Submit & continue
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExercisePreview using step data matches snapshot with overlay and selected false 1`] = `
<div
  className="sc-ksZaOG gLzA-Dq"
>
  <div
    className="sc-gsnTZi dxdISE"
    data-task-step-id={1}
  >
    <div
      className="sc-bczRLJ eAXREJ exercise-step sc-breuTD dSaRVj sc-hAZoDl sc-bjUoiL fZGyZE hmjJCZ preview-card is-selected"
    >
      <div
        onBlur={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        tabIndex={0}
      >
        <div
          id="step-card"
        >
          <div
            className="sc-dkzDqf eYRLRg step-card-header"
          >
            <div>
              <h2
                className="question-info"
              >
                <span>
                  Question 1
                </span>
                <span
                  className="separator"
                >
                  |
                </span>
                <span
                  className="question-id"
                >
                  ID: 
                  1@1
                </span>
              </h2>
            </div>
          </div>
          <div
            className="sc-hKMtZM bwRTaS"
          >
            <div>
              <div
                className="step-card-body exercise-context"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Context",
                  }
                }
              />
              <div
                className="step-card-body exercise-stimulus"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<b>Stimulus HTML</b>",
                  }
                }
              />
              <div
                data-test-id="student-exercise-question"
              >
                <div
                  className="sc-iBkjds bohpoA openstax-question step-card-body"
                  data-question-number={1}
                  data-test-id="question"
                >
                  <div
                    className="answers-table"
                  >
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled"
                      >
                        <input
                          checked={false}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-0"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-0"
                        >
                          <span
                            aria-label="Choice A:"
                            className="answer-letter-wrapper"
                            data-answer-choice="A"
                            data-test-id="answer-choice-A"
                          />
                          <div
                            className="answer-answer"
                          >
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "True",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled"
                      >
                        <input
                          checked={false}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-1"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-1"
                        >
                          <span
                            aria-label="Choice B:"
                            className="answer-letter-wrapper"
                            data-answer-choice="B"
                            data-test-id="answer-choice-B"
                          />
                          <div
                            className="answer-answer"
                          >
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "False",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                  </div>
                  <div
                    className="detailed-solution"
                  >
                    <div
                      className="header"
                    >
                      Detailed solution:
                    </div>
                    <div
                      className="solution"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<span data-math='e^{\\\\text{Blue}}'></span><span data-math='e^{\\\\text{Blue}}'></span>",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="sc-ftvSup fUvOut step-card-footer"
                >
                  <div
                    className="step-card-footer-inner"
                  >
                    <div
                      className="points"
                      role="status"
                    >
                      <span
                        className="attempts-left"
                      />
                    </div>
                    <div
                      className="controls"
                    >
                      <button
                        className="sc-jSMfEi hsxEPT"
                        data-test-id="submit-answer-btn"
                        disabled={true}
                        onClick={[Function]}
                      >
                        Submit & continue
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExercisePreview using step data matches snapshot with overlay and selected true 1`] = `
<div
  className="sc-ksZaOG gLzA-Dq"
>
  <div
    className="sc-gsnTZi dxdISE"
    data-task-step-id={1}
  >
    <div
      className="sc-bczRLJ eAXREJ exercise-step sc-breuTD dSaRVj sc-hAZoDl sc-fnykZs fZGyZE jiPToZ preview-card is-selected"
    >
      <div
        onBlur={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        tabIndex={0}
      >
        <div
          id="step-card"
        >
          <div
            className="sc-dkzDqf eYRLRg step-card-header"
          >
            <div>
              <h2
                className="question-info"
              >
                <span>
                  Question 1
                </span>
                <span
                  className="separator"
                >
                  |
                </span>
                <span
                  className="question-id"
                >
                  ID: 
                  1@1
                </span>
              </h2>
            </div>
          </div>
          <div
            className="sc-hKMtZM bwRTaS"
          >
            <div>
              <div
                className="step-card-body exercise-context"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Context",
                  }
                }
              />
              <div
                className="step-card-body exercise-stimulus"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<b>Stimulus HTML</b>",
                  }
                }
              />
              <div
                data-test-id="student-exercise-question"
              >
                <div
                  className="sc-iBkjds bohpoA openstax-question step-card-body has-correct-answer"
                  data-question-number={1}
                  data-test-id="question"
                >
                  <div
                    className="answers-table"
                  >
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled answer-selected answer-correct"
                      >
                        <input
                          checked={true}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-0"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-0"
                        >
                          <span
                            aria-label="Selected Choice A:"
                            className="answer-letter-wrapper"
                            data-answer-choice="A"
                            data-test-id="answer-choice-A"
                          />
                          <div
                            className="answer-answer"
                          >
                            <div
                              className="sc-gKXOVf QPaPs"
                            >
                              Correct
                               Answer
                            </div>
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "True",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled"
                      >
                        <input
                          checked={false}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-1"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-1"
                        >
                          <span
                            aria-label="Choice B:"
                            className="answer-letter-wrapper"
                            data-answer-choice="B"
                            data-test-id="answer-choice-B"
                          />
                          <div
                            className="answer-answer"
                          >
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "False",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                  </div>
                  <div
                    className="detailed-solution"
                  >
                    <div
                      className="header"
                    >
                      Detailed solution:
                    </div>
                    <div
                      className="solution"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<span data-math='e^{\\\\text{Blue}}'></span><span data-math='e^{\\\\text{Blue}}'></span>",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="sc-ftvSup fUvOut step-card-footer"
                >
                  <div
                    className="step-card-footer-inner"
                  >
                    <div
                      className="points"
                      role="status"
                    >
                      <span
                        className="attempts-left"
                      />
                    </div>
                    <div
                      className="controls"
                    >
                      <button
                        className="sc-jSMfEi hsxEPT"
                        data-test-id="submit-answer-btn"
                        disabled={true}
                        onClick={[Function]}
                      >
                        Submit & continue
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExercisePreview using step data matches snapshot without overlay 1`] = `
<div
  className="sc-ksZaOG gLzA-Dq"
>
  <div
    className="sc-gsnTZi dxdISE"
    data-task-step-id={1}
  >
    <div
      className="sc-bczRLJ eAXREJ exercise-step sc-breuTD dSaRVj sc-hAZoDl sc-idiyUo fZGyZE kuBlnY preview-card "
    >
      <div>
        <div
          id="step-card"
        >
          <div
            className="sc-dkzDqf eYRLRg step-card-header"
          >
            <div>
              <h2
                className="question-info"
              >
                <span>
                  Question 1
                </span>
                <span
                  className="separator"
                >
                  |
                </span>
                <span
                  className="question-id"
                >
                  ID: 
                  1@1
                </span>
              </h2>
            </div>
          </div>
          <div
            className="sc-hKMtZM bwRTaS"
          >
            <div>
              <div
                className="step-card-body exercise-context"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Context",
                  }
                }
              />
              <div
                className="step-card-body exercise-stimulus"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<b>Stimulus HTML</b>",
                  }
                }
              />
              <div
                data-test-id="student-exercise-question"
              >
                <div
                  className="sc-iBkjds bohpoA openstax-question step-card-body has-correct-answer"
                  data-question-number={1}
                  data-test-id="question"
                >
                  <div
                    className="answers-table"
                  >
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled answer-selected answer-correct"
                      >
                        <input
                          aria-details="feedback-1234@5-0"
                          checked={true}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-0"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-0"
                        >
                          <span
                            aria-label="Selected Choice A:"
                            className="answer-letter-wrapper"
                            data-answer-choice="A"
                            data-test-id="answer-choice-A"
                          />
                          <div
                            className="answer-answer"
                          >
                            <div
                              className="sc-gKXOVf QPaPs"
                            >
                              Correct
                               Answer
                            </div>
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "True",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                    <aside
                      className="question-feedback bottom"
                      id="feedback-1234@5-0"
                    >
                      <div
                        aria-label="Answer Feedback"
                        className="arrow"
                      />
                      <aside>
                        <div
                          className="question-feedback-content has-html"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Feedback",
                            }
                          }
                        />
                      </aside>
                    </aside>
                    <div
                      className="openstax-answer"
                    >
                      <section
                        className="answers-answer disabled"
                      >
                        <input
                          checked={false}
                          className="answer-input-box"
                          disabled={true}
                          id="1234@5-option-1"
                          name="1234@5-options"
                          onChange={[Function]}
                          type="radio"
                        />
                        <label
                          className="answer-label"
                          htmlFor="1234@5-option-1"
                        >
                          <span
                            aria-label="Choice B:"
                            className="answer-letter-wrapper"
                            data-answer-choice="B"
                            data-test-id="answer-choice-B"
                          />
                          <div
                            className="answer-answer"
                          >
                            <span
                              className="answer-content"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "False",
                                }
                              }
                            />
                          </div>
                        </label>
                      </section>
                    </div>
                  </div>
                  <div
                    className="detailed-solution"
                  >
                    <div
                      className="header"
                    >
                      Detailed solution:
                    </div>
                    <div
                      className="solution"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<span data-math='e^{\\\\text{Blue}}'></span><span data-math='e^{\\\\text{Blue}}'></span>",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="sc-ftvSup fUvOut step-card-footer"
                >
                  <div
                    className="step-card-footer-inner"
                  >
                    <div
                      className="points"
                      role="status"
                    >
                      <span
                        className="attempts-left"
                      />
                      <div>
                        <strong>
                          Detailed solution:
                        </strong>
                         
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<span data-math='e^{\\\\text{Blue}}'></span>",
                            }
                          }
                        />
                      </div>
                    </div>
                    <div
                      className="controls"
                    >
                      <button
                        className="sc-jSMfEi hsxEPT"
                        data-test-id="submit-answer-btn"
                        disabled={true}
                        onClick={[Function]}
                      >
                        Submit & continue
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
